drivers/mfd/mfd_axp192.c

```diff
@@ -355,18 +387,25 @@ int mfd_axp192_gpio_func_ctrl(const struct device *dev, const struct device *cli
 		}
 		break;
 
+	case 5U:
+		/* EXTEN is an output only pin */
+		break;
+
 	default:
 		ret = -EINVAL;
 	}
+
 	if (ret != 0) {
 		LOG_ERR("Invalid function (0x%x) for gpio %d", func, gpio);
 		return ret;
 	}
 
-	ret = i2c_reg_update_byte_dt(&(config->i2c), gpio_reg_desc[gpio].reg,
-				     gpio_reg_desc[gpio].mask, reg_cfg);
-	if (ret != 0) {
-		return ret;
+	if (gpio < ARRAY_SIZE(gpio_reg_desc)) {
```

Comment:
Could you return earlier if the user asks for an invalid channel?

Reply:
That's not possible here. The write operation can only be executed after the switch-case above has been evaluated. The if is needed to filter the new EXT_EN gpio, that doesn't require a configuration to be written.